docs/tutorial.md

```diff
@@ -550,6 +550,34 @@ the scheduler starts up, it detects that it is already running 3
 `data` tasks and starts 2 more in order to reach the goal state of 5.
 We should observe two more tasks starting, `data-3` and `data-4`.
 
+## Task Configuration management
+
+While the above describes configuration of the scheduler itself via
+environment variables, there's also a need for configuration of the
+underlying service tasks themselves in a flexible way.
+
+To simplify the common task of getting user-facing configuration to
+service tasks, the developer may follow the following convention in
+naming the environment variables for these configuration options:
+`TASKCFG_<TASK_TYPE>_<NAME>`
+
+For example, an option for tasks of type `index-mgr` should be named
+`TASKCFG_INDEX_MGR_FOO`, while an option for tasks of type `data`
```

Comment:
Gave more detail on the conversion and updated the examples to better demonstrate the behavior.

Updated the implementation to convert all non-alphanumerics in task types to underscores (to fit envvar format)